staging/src/scala/quoted/staging/QuoteDriver.scala

```diff
@@ -67,29 +67,25 @@ private class QuoteDriver(appClassloader: ClassLoader) extends Driver {
     ctx.setReporter(new ThrowingReporter(ctx.reporter))
   }
 
-  /** Attempt to recreate a classpath from a classloader.
+  /** Attempt to recreate a classpath from a classloader or the `java.class.path` property
    *
    *  BEWARE: with exotic enough classloaders, this may not work at all or do
    *  the wrong thing.
    */
   private def classpathFromClassloader(cl: ClassLoader): String = {
-    @tailrec
-    def loop(cl: ClassLoader, suffixClasspath: String): String =
-      cl match {
-        case cl: URLClassLoader =>
-          val updatedClasspath = cl.getURLs
-            .map(url => Paths.get(url.toURI).toAbsolutePath.toString)
-            .mkString(
-              "",
-              File.pathSeparator,
-              if (suffixClasspath.isEmpty) "" else File.pathSeparator + suffixClasspath
-            )
-          loop(cl.getParent, updatedClasspath)
-        case _ =>
-          suffixClasspath
-      }
-
-    loop(cl, "")
+    val classpathBuff = List.newBuilder[String]
+    def collectClassLoaderPaths(cl: ClassLoader): Unit = cl match {
+      case cl: URLClassLoader =>
+        collectClassLoaderPaths(cl.getParent)
+        classpathBuff ++=
+          cl.getURLs.iterator.map(url => Paths.get(url.toURI).toAbsolutePath.toString)
+      case _ =>
+    }
+    collectClassLoaderPaths(cl)
+    val systemClasspath = System.getProperty("java.class.path")
+    if (!systemClasspath.isEmpty)
```

Comment:
This is wrong, it's going to pollute the classpath with whatever is on java.class.path which might be totally unrelated to the code we're running.